src/runtime/opencl/opencl_module.cc

```diff
@@ -185,11 +187,15 @@ cl_kernel OpenCLModuleNode::InstallKernel(cl::OpenCLWorkspace* w,
     // create program
     if (fmt_ == "cl") {
       if (program_ == nullptr) {
+#ifndef AOCL_BOARD_NAME
         const char* s = data_.c_str();
         size_t len = data_.length();
         cl_int err;
         program_ = clCreateProgramWithSource(w->context, 1, &s, &len, &err);
         OPENCL_CHECK_ERROR(err);
+#else
+        OfflineCompile(w, t);
```

Comment:
Should be moved to codegen since compiling OpenCL codes for FPGAs takes very long time.

Reply:
Offline compilation was moved to codegen. See also BuildAOCL().